cv.typ

```diff
@@ -2,341 +2,391 @@
+
+        block(width: 100%, breakable: isbreakable, {
+          // line 1: certificate name
+          line_one(cert.name, url: cert.url)
+
+          linebreak()
+          // line 2: issuer and date
+          line_two(cert.issuer, right_end: date)
+        })
+      }
+    })
+  }
 }
 
 #let cvpublications(info, isbreakable: true) = {
-    if info.publications != none {block[
-        == Research & Publications
-        #for pub in info.publications {
-            // parse ISO date strings into datetime objects
-            let date = utils.strpdate(pub.releaseDate)
-            // create a block layout for each publication entry
-            block(width: 100%, breakable: isbreakable)[
-                // line 1: publication title
-                #if pub.url != none [
-                    *#link(pub.url)[#pub.name]* \
-                ] else [
-                    *#pub.name* \
-                ]
-                // line 2: publisher and date
-                Published on #text(style: "italic")[#pub.publisher]  #h(1fr) #date \
-            ]
-        }
-    ]}
+  if info.at("publications", default: none) != none {
+    block({
+      [== Research & Publications]
+
+      for pub in info.publications {
+        let url = pub.at("url", default: none)
+        let date = utils.str_to_date(pub.releaseDate)
+
+        block(width: 100%, breakable: isbreakable, {
+          // line 1: publication title
+          line_one(pub.name, url: url)
+
+          linebreak()
+          // line 2: publisher and date
+          [Published on #line_two(pub.publisher, right_end: date)]
+        })
+      }
+    })
+  }
 }
 
 #let cvskills(info, isbreakable: true) = {
-    if (info.languages != none) or (info.skills != none) or (info.interests != none) {block(breakable: isbreakable)[
-        == Skills, Languages, Interests
-        #if (info.languages != none) [
-            #let langs = ()
-            #for lang in info.languages {
-                langs.push([#lang.language (#lang.fluency)])
-            }
-            - *Languages*: #langs.join(", ")
-        ]
-        #if (info.skills != none) [
-            #for group in info.skills [
-                - *#group.category*: #group.skills.join(", ")
-            ]
-        ]
-        #if (info.interests != none) [
-            - *Interests*: #info.interests.join(", ")
-        ]
-    ]}
+  if (info.at("languages", default: none) != none) or (info.at("skills", default: none) != none) or (info.at("interests", default: none) != none) {
+    block(breakable: isbreakable, {
+      [== Skills, Languages, Interests]
+
+      let langs = info.at("languages", default: none)
+      let langs_join = ()
+      for lang in langs {
+        langs_join.push([#lang.language (#lang.fluency)])
+      }
+
+      dot_list("Languages", langs_join)
+
+      let skills = info.at("skills", default: none)
+      for skill in skills {
+        dot_list(skill.category, skill.skills)
+      }
```

Comment:
To reduce confusion, might I suggest to use the variable name `group`, `skill-group`, or `skill_group`?

```suggestion
      for skill_group in skills {
        dot_list(skill_group.category, skill_group.skills)
      }
```